pkgs/pkgs-lib/formats.nix

```diff
@@ -517,4 +517,37 @@ rec {
     '') {};
   };
 
+  xml = {}: json {} // {
+    type = let
+      valueType = oneOf [
+        bool
+        int
+        float
+        str
+        path
+        (attrsOf valueType)
+        (listOf valueType)
+      ] // {
+        description = "XML value";
+      };
+    in valueType;
```

Comment:
to be honest: i just copied and pasted this from the json... not sure if this is a problem